src/client/app/components/admin/LogMsgComponent.tsx

```diff
@@ -0,0 +1,375 @@
+		setModalLogMessage(logMessage);
+		setModalOpen(true);
+	};
+
+	// Handle checkbox change for log type in the table
+	const handleTableCheckboxChange = (logType: string) => {
+		if (selectedTableLogTypes.includes(logType)) { // Remove log type if already selected
+			setSelectedTableLogTypes(selectedTableLogTypes.filter(type => type !== logType));
+		} else { // Add log type if not selected
+			setSelectedTableLogTypes([...selectedTableLogTypes, logType]);
+		}
+	};
+
+	// Handle checkbox change for log type in the update log
+	const handleUpdateCheckboxChange = (logType: string) => {
+		if (selectedUpdateLogTypes.includes(logType)) {  // Remove log type if already selected
+			setSelectedUpdateLogTypes(selectedUpdateLogTypes.filter(type => type !== logType));
+		} else { // Add log type if not selected
+			setSelectedUpdateLogTypes([...selectedUpdateLogTypes, logType]);
+		}
+	};
+
+	// React effect to keep track of the "Select All" checkbox state
+	React.useEffect(() => {
+		selectedUpdateLogTypes.length === logTypes.length ? setSelectAllUpdate(true) : setSelectAllUpdate(false);
+		selectedTableLogTypes.length === logTypes.length ? setSelectAllTable(true) : setSelectAllTable(false);
+	}, [selectedUpdateLogTypes, selectedTableLogTypes]);
+
+	// Handle "Select All" checkbox change in the table
+	const handleTableSelectAll = () => {
+		selectAllTable ? setSelectedTableLogTypes([]) : setSelectedTableLogTypes(logTypes);
+		setSelectAllTable(!selectAllTable);
+	};
+	// Handle "Select All" checkbox change in the update log
+	const handleUpdateSelectAll = () => {
+		selectAllUpdate ? setSelectedUpdateLogTypes([]) : setSelectedUpdateLogTypes(logTypes);
+		setSelectAllUpdate(!selectAllUpdate);
+	};
+	// Handle sorting of logs by date
+	const handleDateSort = () => {
+		const newDateSortOrder = dateSortOrder === 'asc' ? 'desc' : 'asc';
+		const sortedLogs = [...logs].sort((a, b) => {
+			const dateA = new Date(a.logTime);
+			const dateB = new Date(b.logTime);
+
+			if (newDateSortOrder === 'asc') {
+				return dateA.getTime() - dateB.getTime();
+			} else {
+				return dateB.getTime() - dateA.getTime();
+			}
+		});
+		setDateSortOrder(newDateSortOrder);
+		setLogs(sortedLogs);
+	};
+	// Handle date range change
+	const handleDateRangeChange = (range: Value) => {
+		setLogDateRange(dateRangeToTimeInterval(range));
+	};
+	// Handle page change for pagination
+	const handlePageChange = (newPage: number) => {
+		setCurrentPage(newPage);
+	};
+	// Toggle dropdown for type in the table
+	const toggleTypeTable = () => {
+		setTypeTableDropdown(!typeTableDropdown);
+	};
+	// Toggle dropdown for type in the update log
+	const toggleUpdateLog = () => {
+		setUpdateLogDropdown(!updateLogDropdown);
+	};
+	// Handle showing all logs instead of paginated
+	const handleShowAllLogs = () => {
+		setShowAllLogs(!showAllLogs);
+	};
+
+	// Filter logs based on selected log types and date range
+	const filteredLogs = logs.filter(log => {
```

Comment:
I'm finding the dual use of the date picker for the update and display to be a little confusing. For type and number to display the update area does not impact the logs shown until update is hit. For the date picker it does. I think that difference may confuse users. If you agree then I'm okay with removing this filtering (since can update the logs quickly) or adding it to the log table area (maybe in the Log Time column header?) as a separate date picker.

Reply:
This filteredLogs was previously used for log time after the data is fetched. I forgot to remove it after our discussion about the UI of the page.

I think I will remove it since currently the behavior of Log Time column header is sorting the date after every click. I tried to add the filter in the column header but then every time I clicked on the date range picker or picked a date range it would active the sorting again. Please let me know if later on you'd like to have the filter back.